app/controllers/counselor/counselors_controller.rb

```diff
@@ -10,7 +10,25 @@ def edit
   end
 
   def update
-    puts "GOT INTO UPDATE"
+    @counselor = Counselor.find_by(user_id: current_user.id)
+    @user = User.find(@counselor.user_id)
```

Comment:
Tried it multiple times with no crashing up until now; it's looking for an `id` field within `users` and `find` is what's usually used for that.